test/test_fmm.py

```diff
@@ -485,11 +485,7 @@ def test_sumpy_fmm_timing_data_collection(ctx_factory, use_fft, visualize=False)
             fmm_level_to_order=lambda kernel, kernel_args, tree, lev: order)
     from boxtree.fmm import drive_fmm
 
-    timing_data = {}
-    pot, = drive_fmm(actx, wrangler, (weights,), timing_data=timing_data)
-    logger.info("timing_data:\n%s", timing_data)
-
-    assert timing_data
+    pot, = drive_fmm(actx, wrangler, (weights,))
```

Comment:
Hmm. What's the recommended way to get the timing data?

Reply:
Not quite sure. I briefly talked about that with @inducer and it should be baked into the array context somehow so it doesn't clutter the wrangler interface.

However, that probably needs a bit of effort to get back into a working state, so not quite sure what a timeline would be..

Reply:
Just to write out some questions about that to get an idea (from you and @inducer).

* Do you rely on that interface to get results?
* Can this be merged without it?
* Who's going to do the work to put it back? :cry:

Reply:
> Do you rely on that interface to get results?

Yes, I use it to get results for a paper.

> Who's going to do the work to put it back?

I can try if there's an outline of how to reintroduce it. I have not much experience on arraycontexts, so I have no idea what needs to be done.

Reply:
> I can try if there's an outline of how to reintroduce it. I have not much experience on arraycontexts, so I have no idea what needs to be done.

Yeah, I don't have a good overview either, so that probably needs some discussion. There's a profiling array context here
https://github.com/illinois-ceesd/mirgecom/blob/f9cf0f3557b7d5c89f58e8362ebf50820261219f/mirgecom/profiling.py
but not sure if a similar interface will work for gather timings on the FMM.

> Yes, I use it to get results for a paper.

Ok, then definitely either revert or hold off on merging.

At the moment, the timing code and the manual event handling are not particularly compatible with the `arraycontext` stuff (as it is) and removing both is pretty breaking.